File: experiments/runs/2026-09-17_00-24-45_claim-office-example-mapping_exact-ptdd-v1.1-refactor-subagent-pi_gpt-5-6-sol-codex-4/src/insurance-valuation.ts
const INSURANCE_VALUE_BY_ITEM_TYPE: Record<string, number> = {
  sword: 1000,
  amulet: 600,
  staff: 800,
  potion: 400,
  rune: 250,
  moonstone: 250,
};

export interface InsuredItem {
  type: string;
}

function insuranceValueOf(item: InsuredItem): number {
  return INSURANCE_VALUE_BY_ITEM_TYPE[item.type];
}

export function calculateInsuranceSum(items: InsuredItem[]): number {
  return items.reduce((sum, item) => sum + insuranceValueOf(item), 0);
}

File: experiments/runs/2026-09-20_08-53-35_claim-office-java-example-mapping_exact-ptdd-v1.1-refactor-subagent-pi_gpt-5-6-sol-codex/src/main/java/InsuranceValuation.java
final class InsuranceValuation {
    private InsuranceValuation() {
    }

    static int forItemType(String itemType) {
        return switch (itemType) {
            case "sword" -> 1000;
            case "amulet" -> 600;
            case "staff" -> 800;
            case "potion" -> 400;
            case "rune", "moonstone" -> 250;
            default -> throw new IllegalArgumentException("Unknown item type: " + itemType);
        };
    }
}
